analyzers/src/SonarAnalyzer.Common/Trackers/AssignmentFinder.cs

```diff
@@ -43,7 +43,7 @@ public SyntaxNode FindLinearPrecedingAssignmentExpression(string identifierName,
 
                 foreach (var statement in current.Parent.ChildNodes().TakeWhile(x => x != current).Reverse())
                 {
-                    if (IsAssignmentToIdentifier(statement, identifierName, out var right))
+                    if (IsAssignmentToIdentifier(statement, identifierName, false, out var right))
```

Comment:
I think with this change this method needs some comments describing what it does, and why. Some simple ones to give a direction (example) on what to look at

like give a sample of code where it will find one, and give a sample of code where it won't (depending on the assignment types)

Reply:
I've added a documentation comment on the `IsAssignmentToIdentifier` method for the `anyAssignmentKind` argument. That should make it clear.